### PR TITLE
test(quantic): Add tests for feedback modal edge cases in quanticSmartSnippet e2e tests

### PR DESCRIPTION
**Pull request by SimonMilord**
[SFINT-5933](https://coveord.atlassian.net/jira/software/c/projects/SFINT/boards/440/backlog?assignee=62e7739dec6b328032f09736&selectedIssue=SFINT-5933)

## IN THIS PR:

- Added the following 2 edge cases tests in the smart snippet E2E tests in playwright:

1- when trying to open the feedback modal after executing a query that gave a new answer, should open the feedback modal
2- when trying to open the feedback modal after executing the same query, should not allow to open the feedback modal

I check the presence of the explain why button which opens the feedback modal, this was the check in the Cypress ones.

## TESTS:
image

[SFINT-5933]: https://coveord.atlassian.net/browse/SFINT-5933?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/e2e/quanticSmartSnippet.e2e.ts [hunk 1]
```diff
@@ -113,6 +114,55 @@ useCaseTestCases.forEach((useCase) => {
...
+
+          await smartSnippet.clickFeedbackModalDoneButton();
+
+          const searchResponsePromise = search.waitForSearchResponse();
+          await search.performSearch();
+          await searchResponsePromise;
+
+          expect(smartSnippet.dislikeButton).not.toBeNull();
+          await smartSnippet.clickDislikeButton();
+          expect(smartSnippet.explainWhyButton).not.toBeVisible();
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/e2e/quanticSmartSnippet.e2e.ts:136]: Failed tests?

**Review by erocheleau — COMMENTED** [review]
Looks perfect, just a bunch of red lines for failing tests?

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/e2e/quanticSmartSnippet.e2e.ts (hunk 1 above)
Comment (SimonMilord) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/e2e/quanticSmartSnippet.e2e.ts:136]: uh? 👀

Comment (SimonMilord) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippet/e2e/quanticSmartSnippet.e2e.ts:136]: not sure why, it says the [run](https://github.com/coveo/ui-kit/actions/runs/13144715023) is 🟢 and all the tests are passing locally after running them multiple times:
image

**Comment by mmitiche** [comment]
@SimonMilord  the PR title should be changed to `test(quantic):...`

**Review by mmitiche — APPROVED** [review]
LGTM!

**Review by erocheleau — APPROVED** [review]
